python_modules/dagster/dagster/_core/definitions/run_request.py

```diff
@@ -276,6 +321,24 @@ def partition_key_range(self) -> Optional[PartitionKeyRange]:
         else:
             return None
 
+    @classmethod
+    def for_asset_graph_subset(
+        cls,
+        asset_graph_subset: "AssetGraphSubset",
+        tags: Optional[Mapping[str, str]],
+        title: Optional[str],
+        description: Optional[str],
+    ) -> BackfillDaemonRequest:
+        """Constructs a BackfillDaemonRequest from an AssetGraphSubset. When processed by the sensor
+        daemon, this will launch a backfill instead of a run.
+
+        Note: This constructor is intentionally left private since AssetGraphSubset is not part of the
+        public API. Other constructor methods will be public
+        """
+        return BackfillDaemonRequest(
```

Comment:
In this constructor method (and other future constructors) we could do some introspection on the provided asset_graph_subset to determine if the request can be executed in a single run (and if so return a RunRequest)

I'm inclined to punt on doing this for now since it's not simple to do this introspection efficiently. i think it could also make sense to have it always emit backfills (even if it could be a single run) during the initial rollout since DA will be the first user. If it's always a backfill, then the backfills page becomes the source of truth of what was launched while the new runs page is developed, rather than having some backfills and some single runs

figuring out how to determine if a request should be a single run or a backfill efficiently will be important for future backfill and run consolidation work, so i do plan to spend some time looking in to how to do this, i just dont think it should block this pr